### PR TITLE
Basic SCH implementation

### PR DESCRIPTION
**Pull request by xiankai**
Used the SMN implementation as a reference to add 3 modules:
- Aetherflow
  - Changed implementation to check against a 45 second cooldown
- DoTs
  - Replace with SCH-specific DoTs (Miasma, Bio II)
- Shadow Flare
  - Copied as is

### CODE REVIEW

File: src/parser/jobs/sch/Aetherflow.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			requirements: [
+				new Requirement({
+					name: <Fragment><ActionLink {...ACTIONS.AETHERFLOW} /> cooldown uptime</Fragment>,
+					percent: (this.getQuickenedAetherflowOnCooldown() / (this.parser.fightDuration - FIRST_FLOW_TIMESTAMP)) * 100,
+				}),
+			],
+		}))
+	}
+
+	getQuickenedAetherflowOnCooldown() {
```
Comment (ackwell) [inline src/parser/jobs/sch/Aetherflow.js]: Should keep it using `this.cooldowns.getTimeOnCooldown`.

You can call `this.cooldowns.reduceCooldown` when players cast CDR skills, it'll automatically calculate the correct time on cooldown once that's implemented.

File: src/parser/jobs/sch/About.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, {Fragment} from 'react'
+import {Icon, Message} from 'semantic-ui-react'
+
+import CONTRIBUTORS from 'data/CONTRIBUTORS'
+import CoreAbout from 'parser/core/modules/About'
+
+export default class About extends CoreAbout {
+	description = <Fragment>
```
Comment (ackwell) [inline src/parser/jobs/sch/About.js]: This file needs to be updated to represent the state of the SCH module group.

File: src/parser/jobs/sch/Aetherflow.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			description: `
+				The level 68 trait, Quickened Aetherflow, reduces your Aetherflow cooldown by 5 seconds after using a single stack of aetherflow.
+				Using all your stacks before the cooldown is up effectively reduces the cooldown from 60 to 45 seconds,
+				which helps with mana management.
+			`,
+			requirements: [
+				new Requirement({
+					name: <Fragment><ActionLink {...ACTIONS.AETHERFLOW} /> cooldown uptime</Fragment>,
+					percent: (this.getQuickenedAetherflowOnCooldown() / (this.parser.fightDuration - FIRST_FLOW_TIMESTAMP)) * 100,
+				}),
```
Comment (ackwell) [inline src/parser/jobs/sch/Aetherflow.js]: I'd add a separate Requirement here to deal with using CDR flow actions ineffectively. Uptime and effective usage are sort of two separate things.

File: src/parser/jobs/sch/DoTs.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+				}),
+			],
+		}))
+
+		// Suggestion for DoT clipping
+		const maxClip = Math.max(...Object.values(this._clip))
+		this.suggestions.add(new Suggestion({
+			icon: ACTIONS.BIO.icon,
+			content: <Fragment>
+				Avoid refreshing DoTs significantly before their expiration, except when at the end of the fight. Unnecessary refreshes risk overwriting buff snapshots, and increase the frequency you&apos;ll need to hardcast your DoTs.
```
Comment (ackwell) [inline src/parser/jobs/sch/DoTs.js]: This suggestion is primarily worded for SMN, where you want to keep DoTs up with Tri-Disaster as much as possible. Might need some tweaks.

**Comment by xiankai** [comment]
After our discussion, I've addressed the following issues:
- Use in-built CDR handling
- Compare total aetherflow casts/possible aetherflow casts (added under requirement, or should it be separated out into a suggestion?)
- Updated contribution
- Text fixes (for DoT overwriting and aetherflow description)

And what is hopefully a more holistic overview of aetherflow usage (could add suggestions or highlight problematic rows later):

image

**Comment by ackwell** [comment]
Thanks for those changes, and that table looks great, nice job.

As I've just posted in `#development`, I'll be requesting a new description in the `about` module for the next PR, please and thankee.

Also for future PRs, the code generating the table could probably do with a bit of simplification - might be worth pulling that block outside the JSX and _maybe_ into a few functions depending on feasibility, just to break it up and make it a bit more digestible.

**Pull request merged**